e2e/app/setup.go

```diff
@@ -626,10 +626,16 @@ func writeMonitorConfig(ctx context.Context, def Definition, logCfg log.Config,
 		return errors.New("monitor must use archive node, no archive node found")
 	}
 
+	solverData, ok := def.Infra.GetInfrastructureData().Instances["solver"]
+	if !ok {
+		return errors.New("no solver instance found")
+	}
+
 	cfg := monapp.DefaultConfig()
 	cfg.PrivateKey = privKeyFile
 	cfg.Network = def.Testnet.Network
 	cfg.HaloCometURL = archiveNode.AddressRPC()
+	cfg.SolverIPAddress = solverData.IPAddress.String()
```

Comment:
rather make this: `cfg.SolverAddr = def.Testnet.SolverAddr` which includes the port.